index.js

```diff
@@ -104,29 +143,56 @@ exports.init = function (sbot, config) {
     if (rpc.id === sbot.id) return // ssb-client connecting to ssb-server
     if (isClient) {
       initialized.promise.then(() => {
-        const opts = { version: 3 }
-        const local = toPull.duplex(ebt.createStream(rpc.id, opts.version, true))
-        const remote = rpc.ebt.replicate(opts, (networkError) => {
-          if (networkError && getSeverity(networkError) >= 3) {
-            console.error('rpc.ebt.replicate exception:', networkError)
-          }
-        })
-        pull(local, remote, local)
+        for (let format in ebts) {
+          const ebt = ebts[format]
+          const opts = { version: 3, format }
+          const local = toPull.duplex(ebt.createStream(rpc.id, opts.version, true))
+
+          // for backwards compatibility we always replicate classic
+          // feeds using existing replicate RPC
+          const replicate = (format === 'classic' ? rpc.ebt.replicate : rpc.ebt.replicateFormat)
+
+          const remote = replicate(opts, (networkError) => {
+            if (networkError && getSeverity(networkError) >= 3) {
+              console.error('rpc.ebt.replicate exception:', networkError)
+            }
+          })
+          pull(local, remote, local)
+        }
       })
     }
   })
 
-  function request (destFeedId, requesting) {
+  function request(destFeedId, requesting) {
     initialized.promise.then(() => {
-      if (!isFeed(destFeedId)) return
-      ebt.request(destFeedId, requesting)
+      formatName = 'classic'
+      for (let format in ebts)
+        if (formats[format].isFeed(destFeedId))
+          formatName = format
+
+      const format = formats[formatName]
+
+      if (!(format && format.isFeed(destFeedId))) return
+      
+      ebts[formatName].request(destFeedId, requesting)
     })
   }
 
-  function block (origFeedId, destFeedId, blocking) {
+  function block(origFeedId, destFeedId, blocking) {
     initialized.promise.then(() => {
-      if (!isFeed(origFeedId)) return
-      if (!isFeed(destFeedId)) return
+      formatName = 'classic'
+      for (let format in ebts)
+        if (formats[format].isFeed(origFeedId))
+          formatName = format
```

Comment:
Okay I've seen the 4 lines above copy-pasted in many places so it seems we could make a helper function, called `detectFormatName(feedId)`. Also I noticed that the `formatName` var wasn't declared (not `var formatName`, not `let formatName`) which makes it a global and that's a bit nasty